Xamarin.Forms.Platform.iOS/PageLifecycleManager.cs

```diff
@@ -4,99 +4,31 @@
 
 namespace Xamarin.Forms.Platform.iOS
 {
-	internal class PageLifecycleManager : IDisposable, IDisconnectable
+	internal class PageLifecycleManager
 	{
-		NSObject _activateObserver;
-		NSObject _resignObserver;
-		bool _disposed;
-		bool _appeared;
 		IPageController _pageController;
 
 		public PageLifecycleManager(IPageController pageController)
 		{
 			_pageController = pageController ?? throw new ArgumentNullException("You need to provide a Page Element");
-
-			_activateObserver = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.DidBecomeActiveNotification, n =>
-			{
-				if (CheckIfWeAreTheCurrentPage())
-					HandlePageAppearing();
-			});
-
-			_resignObserver = NSNotificationCenter.DefaultCenter.AddObserver(UIApplication.WillResignActiveNotification, n =>
-			{
-				if (CheckIfWeAreTheCurrentPage())
-					HandlePageDisappearing();
-			});
-		}
-
-		protected virtual void Dispose(bool disposing)
-		{
-			if (_disposed)
-				return;
-
-			if (disposing)
-			{
-				(this as IDisconnectable).Disconnect();
-			}
-
-			_disposed = true;
-		}
-
-		void IDisconnectable.Disconnect()
-		{
-			if (_activateObserver != null)
-			{
-				NSNotificationCenter.DefaultCenter.RemoveObserver(_activateObserver);
-				_activateObserver = null;
-			}
-
-			if (_resignObserver != null)
-			{
-				NSNotificationCenter.DefaultCenter.RemoveObserver(_resignObserver);
-				_resignObserver = null;
-			}
-
-			HandlePageDisappearing();
-
-			_pageController = null;
-		}
-
-		public void Dispose()
-		{
-			Dispose(disposing: true);
-			GC.SuppressFinalize(this);
 		}
 
 		public void HandlePageAppearing()
 		{
-			if (_appeared)
+			// UISplitViewController ping pongs this event when you minimize an app
+			if (UIApplication.SharedApplication.ApplicationState != UIApplicationState.Active)
 				return;
 
-			_appeared = true;
 			_pageController?.SendAppearing();
 
 		}
 
 		public void HandlePageDisappearing()
 		{
-			if (!_appeared || _pageController == null)
+			if (_pageController == null)
 				return;
 
-			_appeared = false;
 			_pageController.SendDisappearing();
 		}
-
-		public bool Appeared => _appeared;
-
-		bool CheckIfWeAreTheCurrentPage()
```

Comment:
The problem with the a page trying to assess whether it should send appearing is that it has to go pretty far up the chain to know if it should send appearing.  Appearing needs to be top down not bottom up. For example if you have a TabbedPage with nested NavigationPages this causes appearing to fire on non visible TabbedPages because the immediate parent is the NavPage which indicates that it is the current page.